drivers/hmis/app/models/hmis/filter/assessment_filter.rb

```diff
@@ -20,6 +21,10 @@ def with_role(scope)
     with_filter(scope, :type) { scope.with_role(input.type) }
```

Comment:
I wasn't sure about whether to attempt to completely remove this old `type` filter in favor of the newly added filter. It seemed to cause other issues (the route `/client/:id/enrollments/:id/intake` failed, for example), but I didn't look deeply into why, and maybe it's a simple fix.

Reply:
@martha we want to adjust the current filter instead of adding a new one. Sorry that wasn't clear! 
The desired dropdown list should look like the below, except `Custom Assessement` is removed and replaced with one item per Custom Assessment Form Definition.

If `project_id` is provided to the picklist query, it should only show Form Definitions that are "available" in the proejct. If `project_id` is not provided (like when rendered on the Client Assessments page) then all custom assessment types should be listed. Does that make sense? I'll address the question of "available" forms on your other comment.


<img width="341" alt="Screenshot 2024-04-22 at 9 47 49 AM" src="https://github.com/greenriver/hmis-warehouse/assets/5333982/1b205dcc-cd61-45ca-9a59-3c49c23280d5">